pkg/image/registryclient/alternativesources.go

```diff
@@ -0,0 +1,170 @@
+package registryclient
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+
+	"github.com/docker/distribution"
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
```

Comment:
It's not appropriate for image/registryclient to take dependencies directly on openshift/api or openshift/client-go, so at a minimum this needs to be moved to another package and the interface (that exposes only incoming repository and alternates) is what you pass in to the context.

Reply:
moved all ICSP fns to own package and instead of passing to context, pass context to it :shrug:  so no changes in registryclient.